device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -974,6 +993,11 @@ private AmqpsSessionHandler addSessionHandler(DeviceClientConfig deviceClientCon
         {
             amqpsSessionHandler = new AmqpsSessionHandler(deviceClientConfig, this);
         }
+        else
+        {
+            // Since a session is being added for the reconnecting device, it can be removed from the reconnectingDeviceSessionHandlers collection
+            this.reconnectingDeviceSessionHandlers.remove(amqpsSessionHandler);
```

Comment:
This list is of multiplexed devices that are temporarily unregistered for reconnection purposes. Since the addSessionHandler call is used to re-add the unregistered device, now is the appropriate time to remove the device from this list.